### PR TITLE
Fix conversion to OpenShift (invalid DeploymentConfig)

### LINKED ISSUE #124: OpenShift conversoin - invalid DeploymentConfig
When creating (`oc create -f ...`) DeploymentConfig created (`kompose convert --dc`) from this [docker-compose.yml](https://github.com/almighty/almighty-core/blob/master/docker-compose.yml)  I get following error:
```
The DeploymentConfig "" is invalid.

* metadata.name: Required value: name or generateName is required
* spec.template: Required value
* spec.selector: Invalid value: null: selector cannot be empty
```

### LINKED ISSUE #130: services should be first in List
When generating List, Services should be first in the List to make sure they are created fist.

### PR DESCRIPTION
**Pull request by kadel**
fixes #124 by add missing conversion to versioned object when saving to files

I've also added commit that fixes #130

### CODE REVIEW

**Comment by janetkuo** [comment]
Mind adding a test to prevent regression?

**Comment by kadel** [comment]
there are already tests that should catch this, but they are not working properly :-(
see #125

**Comment by surajssd** [comment]
@kadel to address this issue in #73 I added a function called `sortServices` at that time somehow it seems to be out of code :(

This function was created because @ngtuna asked for it https://github.com/skippbox/kompose/pull/94#discussion_r74376944

Link to function in diff 
https://github.com/skippbox/kompose/commit/04d1d655525d88de0544a7554f995001927ccae4#diff-41d801ef80f1858d5e8e9695667e4dafR1302

File: pkg/transformer/kubernetes/k8sutils.go [hunk 1]
```diff
@@ -127,24 +127,38 @@ func PrintList(objects []runtime.Object, opt kobject.ConvertOptions) error {
...
+			}
 
-		// version each object in the list
-		list.Items, err = convertToVersion(list.Items)
-		if err != nil {
-			return err
-		}
+			//make sure that Serices are first in the List
+			switch object.(type) {
+			case *api.Service:
```
Comment (surajssd) [inline pkg/transformer/kubernetes/k8sutils.go]: To simplify this switch you can also use `object.GetObjectKind().GroupVersionKind().Kind == "Service"` to do the checks.

**Comment by kadel** [comment]
> @kadel to address this issue in #73 I added a function called sortServices at that time somehow it seems to be out of code :(
> 
> This function was created because @ngtuna asked for it #94 (comment)
> 
> Link to function in diff
> 04d1d65#diff-41d801ef80f1858d5e8e9695667e4dafR1302

ah, I missed that, I'll move that function to Kubernetes Transformer and use that

**Comment by kadel** [comment]
Heyo, can please someone review this?
@surajssd @ngtuna  @janetkuo ?

**Comment by kadel** [comment]
Heyo, can please someone review this?
@surajssd @ngtuna @janetkuo ?

**Comment by runseb** [comment]
+1 looks ok, but did not test

**Pull request merged**